packages/astro/src/core/endpoint/index.ts

```diff
@@ -91,27 +92,39 @@ export function createAPIContext({
 
 type ResponseParameters = ConstructorParameters<typeof Response>;
 
-export class ResponseWithEncoding extends Response {
-	constructor(body: ResponseParameters[0], init: ResponseParameters[1], encoding?: BufferEncoding) {
-		// If a body string is given, try to encode it to preserve the behaviour as simple objects.
-		// We don't do the full handling as simple objects so users can control how headers are set instead.
-		if (typeof body === 'string') {
-			// In NodeJS, we can use Buffer.from which supports all BufferEncoding
-			if (typeof Buffer !== 'undefined' && Buffer.from) {
-				body = Buffer.from(body, encoding);
+export let ResponseWithEncoding: ReturnType<typeof initResponseWithEncoding>;
+let initResponseWithEncoding = () => {
```

Comment:
Mind adding a "TODO" or "HACK" or something comment up to so we know to remove this once Stackblitz supports Node 18, similar to the comment in `polyfill.ts`?